dotnet/src/Connectors/Connectors.AI.OpenAI/OpenAIKernelBuilderExtensions.cs

```diff
@@ -88,6 +89,32 @@ public static class OpenAIKernelBuilderExtensions
         return builder;
     }
 
+    /// <summary>
+    /// Adds an Azure OpenAI text completion service to the list.
+    /// See https://learn.microsoft.com/azure/cognitive-services/openai for service details.
+    /// </summary>
+    /// <param name="builder">The <see cref="KernelBuilder"/> instance</param>
+    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="openAIClient">Custom <see cref="OpenAIClient"/>.</param>
+    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="setAsDefault">Whether the service should be the default for its type.</param>
+    /// <returns>Self instance</returns>
+    public static KernelBuilder WithAzureTextCompletionService(this KernelBuilder builder,
```

Comment:
Not passing HttpClient as argument to the constructor obtained from HttpClientProvider.GetHttpClient(parameters.Config, httpClient, parameters.Logger) method call won't allow the AzureTextCompletion to use SK retry policy configured by the WithRetryHandlerFactory/SetHttpRetryHandlerFactory methods during Kernel building/construction.

Reply:
@SergeyMenshykh If you use this extension method you pass in the `OpenAIClient` you want the use. The client can configure the `HttpClient` they want to use. I don't think we should modify this.

Reply:
Yes, right, I did not notice it.